algorithms/CPlusPlus/Dynamic-Programming/matrix-chain-multiplication.cpp

```diff
@@ -0,0 +1,56 @@
+// Matrix Chain Multiplicatin - Memorization-Approuch
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// Function for matrix chain multiplication
+int matrixChainMulti(int arr[], int i, int j, vector<vector<int>> dp)
+{
+    if (i >= j)
+        return 0;
+
+    if (dp[i][j] != -1)
+        return dp[i][j];
+
+    int countMin = INT_MAX;
+    for (int k = i; k < j; k++)
+    {
+        int temp_ans = matrixChainMulti(arr, i, k, dp) + matrixChainMulti(arr, k + 1, j, dp) + (arr[i - 1] * arr[k] * arr[j]);
+        if (temp_ans < countMin)
+            countMin = temp_ans;
+    }
+    return dp[i][j] = countMin;
+}
+
+// Driver Code
+int main()
+{
+    // dp table initialized with -1
+    vector<vector<int>> dp(1001, vector<int>(1001, -1));
+
+    int n;
+    cout << "Enter the array size" << endl;
+    cin >> n;
+    int arr[n];
```

Comment:
Avoid using variable length arrays ( VLAs ) in C++. Although some compilers ( like gcc ) support them as non-standard extension, they are not allowed by C++ standards so a compiler is not obligated to do so.

I see you are already using std::vector in your program then you should have not problem using it here also.